src/main/java/prefs/Preferences.java

```diff
@@ -18,13 +17,8 @@
 
     public GlobalConfig global;
 
-    public Preferences(boolean isTestMode) {
-        if (isTestMode) {
-            this.fileHandler = new ConfigFileHandler(DIRECTORY, TEST_CONFIG_FILE);
-        } else {
-            this.fileHandler = new ConfigFileHandler(DIRECTORY, GLOBAL_CONFIG_FILE);
-        }
-
+    public Preferences(Optional<String> configFileName) {
```

Comment:
This is changed to make the class more configurable and so testable in a way without having a explicit test mode variable that branches out testing and application logic. The test mode variable may then encourage further branching logic in other methods of the class in the future.

I think it's still simple enough here but if we have to do too much of these to make things more testable then it's an indication of other problems as discussed [here](http://gmoeck.github.io/2012/07/09/dont-make-your-code-more-testable.html).